### ISSUE TITLE
SSC meeting minutes 2023

### ISSUE THREAD

**Issue by ivanov** [jupyter/software-steering-council-team-compass#2]
[2024 meeting minutes](https://github.com/jupyter/software-steering-council-team-compass/issues/22)

The [meeting notes](https://docs.google.com/document/u/1/d/e/2PACX-1vQ1dJs_QMbOG-cEBcdM_KyLcMbsKcU781araeV-ui1h9HZi8Lbo5Qg2Jl46Q9X2ovQl1YTA1YsRGHC1/pub) are edited via google docs during the meeting ([edit link for SSC members](https://docs.google.com/document/d/1jkeKusQSI-YnQZh9fm6CAmne3qy6EC2tzXAtWuo1hxM/)). Each meeting's notes can then be added as a comment to this issue.

# Table of content

- [Dec 11th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-2096451418)
- [Dec 4th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1839388899)
- [Nov 27th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1828630081)
- [Nov 20h](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1828583470)
- [Nov 13th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1828569432)
- [Nov 6th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1828550334)
- [Oct 30th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1787571708)
- [Oct 23th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1775692748)
- [Oct 16th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1775690781)
- [Oct 9th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1775688693)
- [Oct 2nd](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1775686341) 
- [Sept 25th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1735074762)
- [Sept 18th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1728547679)
- [Sept 11th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1728546298)
- [Sept 4th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1705436077)
- [Aug 21st](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1714036187)
- [Aug 14th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1714035921)
- [Aug 7th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1714035602)
- [July 31st](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1659687246)
- [July 24th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1648115123)
- [July 17th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1640199026)
- [July 10th](https://github.com/jupyter/software-steering-council-team-compass/issues/2#issuecomment-1629237370)

**Comment by ivanov** [comment]
# Monday March 13, 2023


## Attendees

- Frederic Collonval, JupyterLab, QuantStack
- Zach Sailer, Jupyter Server, Apple
- Min, JupyterHub, Simula
- Rick Wagner, Security, UCSD
- Sylvain Corlay, Voilà, QuantStack
- Johan Mabille, Kernels, QuantStack
- Eric Charles, Notebook, Datalayer, Anaconda
- Carol Willing, Standards, Noteable


## Decisions made

-   


## Action items

-   


## Agenda and minutes

- Isabela - Meeting times

  - Mondays and Fridays at 8am Pacific (the time of this meeting) seem to have the most availability. Can we rotate times to make it available to more people?
  - Office hours: scheduling for how long and at what frequency

- Min - meetings

  - Can we avoid having regular meetings and conduct business via async means? Both times are very difficult for me, and I don’t think we should conduct things in any synchronous medium in general across the project.
  - Related: what’s the difference between this meeting and Friday’s meeting?
  - Work asynchronously as much as possible
  - The challenge is that in the “early days”, we need to review quite a few JEPs and get things moving—meetings might help us get this done.
  - If we stay asynchronous, can we use some signal to ensure that everyone is actively reviewing.

- Rick - Suggested standing topics: Review list of JEPS, JEP process until stable.

  - <https://peps.python.org/pep-0000/#introduction> (Python PEPs view of PEPs by Approval status and Implementation status) (Carol)
  - Shepherd can be optional when opened by someone already familiar to the process
  - Should we set a mandatory minimum time for discussion on all new JEP (something like 1 month?) ?
  - We should move forward with JEP 29 (revisit the JEP process). But in parallel we should make progress on some JEPs to revive the evolution as quickly as possible.

- \[Frederic] Incubator organization

- \[Frederic][Markdown syntax evolution](https://github.com/jupyter/software-steering-council-team-compass/issues/1): addition of MermaidJS in[JupyterLab](https://github.com/jupyterlab/jupyterlab/pull/14102) and [nbconvert](https://github.com/jupyter/nbconvert/pull/1957).

- Zach -

  - Who is going to JupyterCon?

    - Frederic
    - Johan
    - Eric
    - Min
    - Sylvain

- \[NEXT ACTIONS]

  - Batch of JEPs for Mar 13 to April 12

    - Open Under Consideration

      - [JEP 66 - Kernels port connection](https://github.com/jupyter/enhancement-proposals/pull/66)

  - Proposed list of JEPs to update status (Smaller team to curate)

    - Abandoned, Withdrawn, and Rejected PEPs
    - Accepted PEPs (accepted; may not be implemented yet)

  - \[Frederic] Incubator - open PR to update the incubator process to align with the new governance and mentioning the process is obsolete and will be revised by the SSC

  - \[Frederic] Markdown syntax evolution - open JEP for it

**Comment by ivanov** [comment]
# Monday April 17, 2023


## Attendees

- Frederic Collonval, JupyterLab
- Rick Wagner, Security
- Itay Dafna, Widgets
- Zach Sailer, Server
- Paul Ivanov, Foundations
- Johan Mabille, Kernels
- Sylvain Corlay, Voilà
- Nick Bolweg


## Decisions made

- call for a vote on the Reviewed JEP Process by end of the week


## Action items

- \[Paul+anyone else] Look at what Python does for PEPs to align the JEP process with that

- \[Johan] Creating jupyter standard GitHub organization (with sync with Carol)

- \[Zach]

  - Email EC about schema.jupyter.org.
  - Submit a JEP for formalization of kernelspec

- \[Johan] Plan to call for a vote on the Reviewed JEP Process by end of the week


## Agenda and minutes

- [Zoom link](https://numfocus-org.zoom.us/j/87208914759?pwd=dm43R25meEYxMll0NllsSlJWUktwQT09)

- Start reviewing[Reviewed JEP process](https://github.com/jupyter/enhancement-proposals/pull/104)

  - Rick: For labels, is it possible to make them not conflict

  - Zack: what happens to the JEP JEP - do we make a new one, or update the old one

    - PEPs have “superseded by” “replaces”
    - Last edited
    - Stick to whatever python does, and do that

  - Nick: look at XMPP model

- Discuss having the protocol documentation in its own repository (and where)

  - Roadmap
  - Tools to generate code off of models/schemas
  - Consider conformance testing to confirm adherence
  - Jupyter standards org with a repo for

- Start reviewing[Kernel handshaking pattern proposal](https://github.com/jupyter/enhancement-proposals/pull/66)

  - Race condition

  - Kernel provisioners are doing this same pattern

  - How do we transition from one to the other

    - Keep this information in the kernelspec

- Zach

  - Grab some domains?

  - events.jupyter.org

    - Let’s keep this reserved for “events” as in social events, not schemas/protocols.

  - protocol.jupyter.org

  - schemas.jupyter.org

    - Move events to this new URI.

  - Who has the authority to make these domains

**Comment by minrk** [comment]
Thanks @ivanov!

**Comment by ivanov** [comment]
# Monday April 24, 2023  
Attendees

- Frederic Collonval, JupyterLab, QuantStack
- Paul Ivanov, Jupyter Foundations, Project Jupyter
- Zach Sailer, Server, Apple
- Sylvain Corlay, Voilà, QuantStack
- Afshin T. Darian, \[EC, guest], QuantStack
- David Brochart, \[guest], QuantStack
- Johan Mabille, Jupyter Kernels, QuantStack


## Decisions made

- none


## Action items

- \[Paul] Open an issue about asynchronous communication
- \[Zach] schema.jupyter.org JEP
- Action items from previous meeting


## Agenda and minutes

- \[Paul] How to include and encourage more asynchronous participation in SSC activities / meetings

  - <https://github.com/jupyter/software-steering-council-team-compass/issues/2>

  - Johan: pinging everyone on every PR seems like overkill

    - Prefer to keep it for votes

  - Zach: there’s not one single channel that we know everyone should be watching

    - Github’s tough

      - Notification bankruptcy

    - Google groups might be it - but no confirmation that someone’s seen it

      - Need an emoji reaction per email

  - Maybe gitter / slack channel

    - Zach - is new gitter painful

      - Johan - no, we use it at QuantStack
      - Darian - might be fine, might fall short (notifications)

  - Discord server?

  - Private vs public conversations

    - Not private, but more internal communication -

  - Try something out

- \[Johan] What level of granularity do we want to give to jupyter-standards repositories?

- \[Johan] Initial documentation of kernel protocol in[jupyter-standards](https://github.com/jupyter-standards/kernel-protocol/pull/1)

- \[Johan][Kernel JSON schema JEP](https://github.com/jupyter/enhancement-proposals/pull/105)

  - Don’t want to open a JEP for each new schema, but this one seems important
  - Motivation for definitions is for auto-generation / interoperability of code, and unit tests

- \[Zach] schema.jupyter.org subdomain

  - Went to the EC office hours to discuss this topic

  - Need to determine governance around this subdomain.

    - Who has the authority to publish schemas here?

    - Does every new schema require a JEP (hopefully not…)

      - Paul: Sub-projects can have their own schemas that don’t require a full JEP process, but cross-project schemas probably should be JEPs.
      - Darian: make a template that asks if this schema will be used in multiple projects, and if so, it falls under the purview of the SSC, but if it is for a single sub-project then anyone with merge rights can merge it.

  - Will submit a JEP this week.

    - Add some namespace to schema URIs to help address the issue of publishing serious stuff too soon, e.g. /ratified/, /experimental/, …
    - Require “versioning” in all URIs.

- \[Paul] did JEP improvements vote start from last week?

  - Will try to do it this week.

- \[Johan] another JEP:[handshake pattern](https://github.com/jupyter/enhancement-proposals/pull/66)

  - Main disagreement about what level of detail on the impact we should have in the JEP

  - Gateway is not using the connection files

    - This proposal is closer to what they are using

**Comment by ivanov** [comment]
# Monday,May 1, 2023

- Paul Ivanov, Foundations, Project Jupyter
- Itay Dafna, Widgets, Netflix
- Rick Wagner, Security, UCSD
- Carol Willing, Standards, Noteable
- Sylvain Corlay, Voilà, QuantStack
- Kevin Bates
- Zach Sailer, Server, Apple


## Decisions made

- None


## Action items

- Another pass on the Kernel Handshake JEP to include the testing tool


## Agenda and minutes

- [Kernel Handshake JEP](https://github.com/jupyter/enhancement-proposals/pull/66) discussion (30 minutes)

  - Notes on<https://hackmd.io/APJMu0puQdKQ4L_Vj2Pf7w>
  - Another pass on the Kernel Handshake JEP to include the testing tool

- (5 minutes)[PyPI Organizations](https://github.com/jupyter/security/issues/61) - Rick, Security

  - Meta-thinking: Details of managing and releasing software should happen at the SSC, even if the EC will be making the decision.
  - How are we maintaining who should have privileged accounts
  - Document a process for asking for a change to be made

- (10 minutes)[ subdomain and repo for schemas under jupyter.org](https://github.com/jupyter/enhancement-proposals/pull/108)

  - Feels like a low-risk JEP, would be good to include it in other JEPs
  - Single repo for all of them.
  - “Hello, world” JEP

- Heads up: lots of follow up conversations on the notebook format

  - JEPs are being written, give them the attention they deserve when being submitted

- [Copy-to-globals support for debugger_info](https://github.com/jupyter/enhancement-proposals/pull/93)

- Shepherd discussion

  - One JEP at a time - assign
  - End of agenda have list of JEPs that were discussed in the meeting and ones that are in-flight

- Software Steering Council Meeting in person next week at JupyterCon

**Comment by fcollonval** [comment]
# Monday, May 15, 2023

* Johan Mabille, Jupyter Kernels, QuantStack
* Itay Dafna, Jupyter Widgets, Netflix

## Decisions made
None

## Action items
None

## Agenda and minutes

Minimal attendance, no discussion/decisions today.

**Comment by fcollonval** [comment]
# Monday, May 22, 2023

* Johan Mabille, Jupyter Kernels, QuantStack
* Itay Dafna, Jupyter Widgets, Netflix
* Afshin Darian, EC

## Decisions made

No decisions made on Carol’s proposal yet. The EC will meet and decide whether there is merit in changing the process or not. In case there isn’t, nothing changes and the existing process remains. In case the EC finds there is more discussion to be had on this topic, the conversation will continue by sending a member of the EC to the SSC calls.

## Action items
None

## Agenda and minutes
* Discussed the JEP review process in [Reviewed JEP process by JohanMabille · Pull Request #104 · jupyter/enhancement-proposals (github.com)](https://github.com/jupyter/enhancement-proposals/pull/104). Discussed Carol’s proposal of overhauling the JEP process such that there is a type of working group with a shepherd picked from the SSC, where the final decision is put to a vote by the EC.
* The main concerns:
   * There is an excess burden on SSC members to make decisions where not all members can meaningfully contribute. 
   * The JEP process is not transparent enough
   * Slow moving-JEP is a good thing, whereas at the moment we’re going faster

**Comment by JohanMabille** [comment]
# Monday, Jun 12, 2023
## Attendees
- Frederic Collonval, QuantStack, JupyterLab
- Itay Dafna, Netflix, Widgets
- Zach Sailer, Apple, Server
- Johan Mabille, QuantStack, Kernel
- Afshin Darian, QuantStack, EC

## Decisions made

None

## Action items

- [Zach] Update the JEP guideline
- [Johan] Call for vote on [Kernel optional features](https://github.com/jupyter/enhancement-proposals/pull/92)
- [Frederic] Call for vote on [$schema key]( https://github.com/jupyter/enhancement-proposals/pull/97) (+ update URL) and [text-based diagram](https://github.com/jupyter/enhancement-proposals/pull/101)
- [Frederic] Push on nbformat move

## Agenda and minutes

- Vote on [Kernel handshaking pattern](https://github.com/jupyter/enhancement-proposals/pull/66)
  - Quorum reached - how to make the vote time bounded to have the ball rolling
  - In the future, we should probably set rules on phases and make the vote period and vote period (aka the discussion had enough time to be done prior to the vote).
Vote is specified as a 2-week period but there is nothing for the other phases
We could give the shepherd the ability to set soft deadlines for phases - probably not written in the rule but depending on the discussion at the SSC.
A minimal period of commitment by the author should be at least 6 weeks to cover the different phases
  - On this specific one, Eric posted a comment asking for additional time to look at it this week. We will take the final decision next week unless there is a major issue raised by Eric.

- Nbformat place - GitHub jupyter-standards?
  - Some concerns raised by Matthias Bussonnier that we should keep the number of GitHub organizations low for security / audit reasons. But otherwise no objections.
  - [Frederic] will push for doing the move

- JEP discussion / review
Add text-based diagram in markdown cells: https://github.com/jupyter/enhancement-proposals/pull/101 
Call for vote
$schema in nb format: https://github.com/jupyter/enhancement-proposals/pull/97
Update the schema to schema.jupyter.org for consistency
Call for vote
Kernel Optional Features https://github.com/jupyter/enhancement-proposals/pull/92 
Add the diff to the kernel spec
Call for vote


Markdown flavor JEP: https://github.com/jupyter/enhancement-proposals/pull/99
Add note about giving 2 weeks more discussion
Markdown based format: https://github.com/jupyter/enhancement-proposals/pull/103
Add note about giving 2 weeks more discussion
Set up of the schema website
One of the blocker is to get the EC to update the DNS

Discussion about the SSC organization
Is the council neutral enough?
Number of attendees is lower than expected
Asynchronous participation is though
Having a core number of people is mandatory for the subjects to be pushed forward - this is already how the various subgroup calls are organized.
To improve credibility, we could add in the JEP of JEP to have a smaller group of groups that analyze a JEP to provide a conclusion to help the members have a better idea of what to vote for. This will help reduce the involvement required.

**Comment by JohanMabille** [comment]
# Monday, Jun 19, 2023

June 19 is a US bank holidy, the meeting was adjourned until next week.

**Comment by fcollonval** [comment]
## Monday, Jul 3, 2023

### Attendees

- Sylvain Corlay, QuantStack, Voilà
- Afshin Darian, QuantStack, EC
- Johan Mabille, QuantStack, Kernel
- Itay Dafna, Netflix, Widgets

### Decisions made

None

### Action items
- Itay to email Paul and Carol to confirm the plan to merge Foundations and Standards
- Check on quorum for JEP 101 next week when we have more more votes

### Agenda and minutes

US public holiday (4th of July)

**Comment by fcollonval** [comment]
## Monday, Jul 10, 2023

### Attendees
- Frederic, QuantStack, JupyterLab
- Itay Dafna, Netflix, Widgets
- Rick Wagner, UCSD, Security

### Decisions made

- [Adding $schema to notebook format](https://github.com/jupyter/enhancement-proposals/pull/97) - Approved

### Action items
- Itay will open a PR to get the two merged on the SSC website.

### Agenda and minutes

- Carol and Paul confirmed they would like Standards and Foundations to merge via a reply to Itay’s email.
- We had a short discussion on the responsibilities of the Standards and Foundations subprojects.
- [JEP 101](https://github.com/jupyter/enhancement-proposals/pull/101) adopt a text-based diagram for markdown
  - Due to the high number of Abstain we will continue to discuss this next week to get a broader audience for this JEP

**Comment by fcollonval** [comment]
## Monday, Jul 17, 2023

### Attendees

- Frederic, QuantStack, JupyterLab
- Rick Wagner, UCSD, Security
- Afshin Darian, QuantStack, EC
- Johan Mabille, QuantStack, Kernel

### Decisions made
- None

### Action items

- None

### Agenda and minutes

- [JEP of JEP](https://github.com/jupyter/enhancement-proposals/pull/104) discussion

**Comment by fcollonval** [comment]
Monday, Jul 24, 2023
Attendees
Martha Cryan, DEI Committee
Frederic, QuantStack, JupyterLab
Johan Mabille, QuantStack, Kernels
Sylvain Corlay, QuantStack, Voilà
Afshin Darian, QuantStack, EC (observing)
Decisions made
None
Action items
None
Agenda and minutes
Martha introduction
[Markdown text-based diagram](https://github.com/jupyter/enhancement-proposals/pull/101): we should provide a way for Nick to defend his case in front of unconvinced members
DNS ownership: Darian is in the process to get domain transfer to a Jupyter account.

**Comment by fcollonval** [comment]
## Monday, Jul 31, 2023
### Attendees
- Afshin Darian, QuantStack, EC (observing)
- Itay Dafna, Netflix, Widgets
- Zach Sailer, Apple, Server
- David Brochart, QuantStack

### Decisions made
None

### Action items
None

### Agenda and minutes
- We had a discussion around the JEP process and the JEP of JEPs.
  - How do we properly distribute/delegate the load of reviewing the backlog of JEPs?
  - Ideally, SSC reps can work with their subproject team members to evaluate the proposals together.
- [Open discussion] Require a (infrequent) synchronous meetings for all SSC representatives.
  - Why? 
    - The SSC is still learning how to operate—synchronous meetings would help us evaluate our process and get better at it.
    - It’s been difficult to build momentum without all 11 members proactively engaging open JEPs together.
  - Possible cadence: once-a-month meeting.
  - Alternate times each month so that it equally inconveniences everyone. :) 
  - It’s important that we do not reduce the importance of the weekly meetings, though.
    - These weekly meetings can become “working” meetings.
- How do we handle a JEP with a small majority vote (with multiple abstentions)?
  - What’s the reason for the abstentions? Are they actually “soft ‘no’s”? 
  - In such a case, we should be cautious about merging the JEP and raise the concern.

**Comment by JohanMabille** [comment]
## Monday, Sept 04, 2023

THis is Labor Day in the US, the meeting was cancelled.

**Comment by fcollonval** [comment]
# Monday, Aug 7, 2023

## Attendees

- Frederic Collonval, QuantStack, JupyterLab
- Zach Sailer, Apple, Jupyter Server
- Itay Dafna, Netflix, Jupyter Widgets
- Afshin T. Darian, QuantStack, EC (observing)

## Decisions made

- None

## Action items

- \[Zach\]: send proposal to improve SSC member role

## Agenda and minutes

- Quick follow-up on synchronous meeting discussion from last week

  - Two threads on the Team Compass—let’s comment there

    - [*https://github.com/jupyter/software-steering-council-team-compass/issues/5*](https://github.com/jupyter/software-steering-council-team-compass/issues/5)
    - [*https://github.com/jupyter/software-steering-council-team-compass/issues/7*](https://github.com/jupyter/software-steering-council-team-compass/issues/7)

**Comment by fcollonval** [comment]
# Monday, Aug 14, 2023

## Attendees

- Frederic Collonval, QuantStack, JupyterLab
- Zach Sailer, Apple, Jupyter Server
- Itay Dafna, Netflix, Jupyter Widgets

## Decisions made

- None

## Action items

- Zach will send an email out to SSC google group to solicit feedback on
  how to improve participation.

## Agenda and minutes

- More discussion on how to improve participation in Jupyter SSC. 

  - We’re seeing low participation from folks on the SSC

  - This is blocking us from making progress on reviewing JEPs.

  - Previously we discussed having a monthly synchronous to ensure all
    members. 

    - The hope was that meeting in person would help develop
      trust/relationship between members and improve the efficiency of
      the council.
    - This is still under open discussion. There are many pros/cons to
      this approach. This discussion is happening in parallel to the
      broader discussion on how to improve participation.

  - No matter the results of the synchronous meeting discussion, we
    **need** to be better at asynchronous communication/work.

  - How do we improve this? 

    - Vocalize some expectations and get folks to agree. This doesn’t
      need to be written into governance docs—it’s self-imposed by the
      council; a little peer pressure ;).
    - Regular check-ins on the google group.

- Define minimum expectations

  - Every member needs to subscribe to and actively respond to:

    - SSC Google Group
    - SSC Team Compass repo
    - JEP repo

  - Expect, at a minimum, to spend 2 hours/week on SSC duties.

    - The synchronous meeting offers a good opportunity to budget this
      time while spending some good time with fellow council members.
    - If you can’t meet these expectations, you might need to evaluate
      if you have enough bandwidth for this council.

  - There is enough traffic on the channels above to spend two
    hours/week.

**Comment by ivanov** [comment]
# Monday, Sept 4, 2023

This was Labor Day in the US, the meeting was canceled.

**Comment by ivanov** [comment]
# Monday, Sep 11, 2023

## Attendees

- Frederic Collonval, QuantStack, JupyterLab
- Zach Sailer, Apple, Server
- Johan Mabille, QuantStack, Kernels
- Afshin T. Darian, QuantStack, EC (observing)
- Rick Wagner, UCSD, Security
- Itay Dafna, Netflix, Jupyter Widgets


## Decisions made

- None


## Action items

- Before moving any JEPs forward, we need to do two things:

  - finish reviewing [JEP process](https://github.com/jupyter/enhancement-proposals/pull/104) to ensure the JEP process is defined.
  - finalize the merge of foundations/standards


## Agenda and minutes

- Joint EC/SSC sync/async meeting discussion

**Comment by ivanov** [comment]
# Monday, Sep 18, 2023

## Attendees

- Frederic Collonval, QuantStack, JupyterLab
- Zach Sailer, Apple, Server
- Sylvain Corlay, QuantStack, Voilà
- Itay Dafna, Netflix, Widgets
- Johan Mabille, QuantStack, Kernels


## Decisions made

-


## Action items

- Reach out to Carol to make sure she sees the merging standards/foundation proposal.

- Add more details to SSC role expectations

  - Weekly time commitment
  - Codify the working call
  - Courtesy to acknowledge posts

- Move Johan’s JEP into voting phase.


## Agenda and minutes

- Announcement

  - Darian got DNS set up: <https://schema.jupyter.org/> 
  - For now it is linked to: <https://github.com/jupyter/schema>

- The previous point raises a good point: should we not use  <https://github.com/jupyter-standards/> and move everything under  <https://github.com/jupyter/> following <https://github.com/jupyter/governance/pull/174> 

  - The conclusion of the discussion is that it makes sense to not create a new org as standards are merging with foundation.

- Follow-up from last week

  - finish reviewing [JEP process](https://github.com/jupyter/enhancement-proposals/pull/104) to ensure the JEP process is defined.
  - finalize the merge of foundations/standards

**Comment by fcollonval** [comment]
# <span id="anchor"></span>Monday, Sep 25, 2023

## Attendees

- Frederic Collonval, QuantStack, JupyterLab
- Sylvain Corlay, QuantStack, Voilà
- Rick Wagner, UCSD, Security
- Johan Mabille, QuantStack, Kernels
- Itay Dafna, Netflix, Jupyter Widgets
- Mike Krassowski (attending as free participant)

## Decisions made

- [*Merge Foundations and Standards
  subprojects*](https://github.com/jupyter/governance/pull/174) : Vote
  has passed let’s move forward on this.

## Action items

- None

## Agenda and minutes

- [*Reviewed JEP
  process*](https://github.com/jupyter/enhancement-proposals/pull/104)

  - Voting period

  - Abstention - discussion about solution:

    - Adding a minimal non-abstain vote to avoid a JEP passing with
      ridiculous vote enforcing the SSC members to get educated to
      reduce the number of abstention  
      Some members point out that it will not lift the concern of having
      a decision passed without a strong confidence.
    - Always enforce simple majority  
      Issue is that abstention is identical to no for the final results
    - Iterate with the subproject council to get a vote status (that
      could still be abstain)

  - Discussion to add new JEP states: 'superseded' and 'active'  
    No counter arguments

- Async or sync discussion  
  We will keep a hybrid approach with non-mandatory sync meeting. But
  votes and discussions should be reported on the issues.

- [*Clarify restart to mean restart
  in-place*](https://github.com/jupyter/enhancement-proposals/pull/118) 

  - What is the process to make the JEP move forward? What is the
    priority?  
    The process and the backlog makes this unclear. But if the JEP is
    short and uncomplex, it can be pushed quickly.

**Comment by Zsailer** [comment]
Monday, Oct 2, 2023
Attendees
Frederic Collonval, QuantStack, JupyterLab
Afshin T. Darian, QuantStack, EC (observing)
Action items
None
Agenda and minutes
No discussion as only one member was attending

**Comment by Zsailer** [comment]
# Monday, Oct 9, 2023
## Attendees
* Frederic Collonval, QuantStack, JupyterLab
* Afshin T. Darian, QuantStack, EC (observing)
## Action items
* None
## Agenda and minutes
* No discussion as only one member was attending

**Comment by Zsailer** [comment]
# Monday, Oct 16, 2023
## Attendees
* Frederic Collonval, QuantStack, JupyterLab
* Afshin T. Darian, QuantStack, EC (observing)
* Johan Mabille, QuantStack, Kernels
* Itay Dafna, Netflix, Widgets
* Zach Sailer, Apple, Server
## Decisions made
* None
## Action items
* None
## Agenda and minutes
* https://github.com/jupyter/enhancement-proposals/pull/104 ready for a vote
* Close https://github.com/jupyter/software-steering-council-team-compass/issues/8
* Close https://github.com/jupyter/software-steering-council-team-compass/issues/5
* Spent part of the call drafting a section for our docs describing how the SSC operates.
  * https://hackmd.io/hdsEqcJ3TEOzxEth6Iwbiw?edit
* schemas repo
  * Use the gh-pages as the canonical place where the schema web pages exist.

**Comment by Zsailer** [comment]
# Monday, Oct 30, 2023
## Attendees

- Zach Sailer, Apple, Server
- Frederic Collonval, QuantStack, JupyterLab
- Afshin T. Darian, QuantStack, EC (observing)
- Johan Mabille, QuantStack, Kernels
- Itay Dafna, Netflix, Widgets
- Gabriel Fouasnon, Quansight Labs, Accessibility

## Action items

- Add Gabriel (@gabalafou) to all the things.
  - [ ] Team compass repo: https://github.com/jupyter/software-steering-council-team-compass
  - [ ] Google Group
  - [ ] Office Hour Calendar invite
  - [ ] Jupyter.org (https://github.com/jupyter/jupyter.github.io)
  - [ ] Voting template: https://github.com/jupyter/software-steering-council-team-compass/issues/3
  - [ ] Team Compass “current members” docs: https://jupyter-software-steering-council-team-compass.readthedocs.io/en/latest/#current-steering-council-members
  - [ ] Add to Github SSC team: https://github.com/orgs/jupyter/teams/software-steering-council
- (Zach) Move 118 to voting
- (Zach) Create a sheet to store contact info for SSC members
- (Zach) open issue to track URI rules for schema.jupyter.org work on project board
- (Whole council) Next JEPs to review:
- https://github.com/jupyter/enhancement-proposals/pull/105
- https://github.com/jupyter/enhancement-proposals/pull/106
- (Gabriel) Open issue to create an SSC alumni page

## Agenda and minutes

- Voting period for https://github.com/jupyter/enhancement-proposals/pull/104 has ended
  - Merging since it is passes 9 yes, 1 abstain.
- Open voting here: https://github.com/jupyter/enhancement-proposals/pull/118
- Should we have a list of “previous/alumni council members”?
  - General consensus of folks in meeting was “yes”.
  - https://github.com/jupyter/software-steering-council-team-compass/pull/18

**Comment by JohanMabille** [comment]
# Monday, November 6, 2023

## Canceled

**Comment by JohanMabille** [comment]
# Monday, November 13th, 2023
## Attendees

- Zach Sailer, Apple, Server
- Frederic Collonval, QuantStack, JupyterLab
- Johan Mabille, QuantStack, Kernels
- Itay Dafna, Netflix, Widgets
- Gabriel Fouasnon, Accessibility

## Action items

- Spend time over this next week re-educating ourselves on this JEP. We’d like to open a vote on this next office hour:  https://github.com/jupyter/enhancement-proposals/pull/93
- Centralize Steering Council source
  - Move member list to Jupyter.org and governance pages
  - Create an “Alumni” section in Jupyter.org
- Reach out to Martha to help onboard.
- Write “onboarding” page for new SSC members
- (Johan) Revising this JEP and we can begin voting on it during the next SSC office hour:
  - [#82 Move kernel_info request on the control channel proposal](https://github.com/jupyter/enhancement-proposals/pull/82)

## Agenda and minutes

- Voting period has ended for [#118 Clarify restart to mean restart in-place](https://github.com/jupyter/enhancement-proposals/pull/118).
  - Merging since it passes 9 yes, 1 abstain.
- [#93 JEP for copy-to-globals support in debugger_info](https://github.com/jupyter/enhancement-proposals/pull/93) can be reviewed during the week and called for a vote next week
- We discussed #91 Add JEP for sub-shells 
  - It’s likely to be re-visited and revised, so hold off on voting.
- [#82 Move kernel_info request on the control channel proposal](https://github.com/jupyter/enhancement-proposals/pull/82)

**Comment by JohanMabille** [comment]
# Monday, November 20th, 2023

## Cancelled

**Comment by JohanMabille** [comment]
# Monday, Nov 27, 2023
## Attendees

- Zach Sailer, Apple, Server
- Frederic Collonval, QuantStack, JupyterLab
- Johan Mabille, QuantStack, Kernels
- Itay Dafna, Netflix, Widgets

## Action items

## Agenda and minutes

- Jupyter Incubation process
  - This is a responsibility of the SSC, but not one we have addressed yet.
  - We looked at [CNCF’s](https://github.com/cncf/toc/blob/main/process/project_proposals.md) approach to project phases, e.g. sandbox→incubating→graduated.
  - We’re trying to avoid having an incubator per subproject.
  - What kind of projects require incubation? Is this for subproject-level status only, i.e. get a vote on the SSC?
  - What about smaller projects that don’t need full Jupyter subproject status?
  - We discussed two possible outcomes for incubation.
      - i. New Jupyter Subproject (with SSC vote)
      - II. New project under something like jupyterlab-contrib (probably renamed to something like jupyter-contrib).
        - Jupyter-contrib has a working group
        - Projects are submitted to the working group for review
        - These projects must follow some standards
          - These projects follow a [checklist like this](https://github.com/jupyterlab-contrib/jupyterlab-contrib.github.io/issues/50).
          - These projects have a standard release process (maybe using jupyter-releaser)
          - The working group has release+merge (npm, PyPI, Github) rights
          - These projects inherit the Jupyter BSD license.
        - These projects get added to a google group where we send updates about possible API changes early enough for these s ubprojects to update.
  - When a person or group wants to bring a codebase, repo, or suborg into the Jupyter organization, we should have a clear path for them to take.

**Comment by JohanMabille** [comment]
# Monday, Dec 4, 2023
## Attendees

Gabriel Fouasnon, Jupyter Accessibility
Sylvain Corlay, Voilà
Johan Mabille, Kernels
Zach Sailer, Server

## Action items

## Agenda and minutes

- Gabriel asks for primer about kernel to better understand JEP [Support kernel_info request on the control channel](https://github.com/jupyter/enhancement-proposals/pull/82)
  - Discussion around how there is a documentation gap here
  - Resources shared
    i. [Messaging in Jupyter](https://jupyter-client.readthedocs.io/en/latest/messaging.html)
    ii. [Jeremy Tuloup, Johan Mabille Navigating the Jupyter Landscape | JupyterCon 2023](https://www.youtube.com/watch?v=uWJ0-OPKTxI)
- Context for [Support kernel_info request on the control channel](https://github.com/jupyter/enhancement-proposals/pull/82)
  - Five channels between kernel and server: heartbeat, IO pub, stdin (kernel asks for input), shell, and control
  - When you want to execute code, you send it on the shell
  - When connecting to a kernel, you send a `kernel_info` request on the shell; but if the shell is busy, e.g. paused on a breakpoint or executing a long job, you have to wait for a long time
- Discuss [JEP for copy-to-globals support in debugger_info #93](https://github.com/jupyter/enhancement-proposals/pull/93)
- Discussed creating a kernel roadmap for JEPs voted on
  - Which version of the protocol *requires* a feature
  - Which version of the protocol is *required* by clients such as jupyter client
  - What happens in case of mis-communication of the supported protocol
  - The use of jupyter-kernel-tests as the main testing tool
- Major efforts that the SSC should take on:
  - Protocol clarification, hardening, and documentation
  - Incubation process
  - Schema publishing

**Comment by gabalafou** [comment]
# Monday, December 4


## Attendees



* Gabriel Fouasnon, Jupyter Accessibility
* Sylvain Corlay, Voilà
* Johan Mabille, Kernels
* Zach Sailer, Server




## Agenda and minutes



* Gabriel asks for primer about kernel to better understand JEP [Support kernel_info request on the control channel](https://github.com/jupyter/enhancement-proposals/pull/82)
    * Discussion around how there is a documentation gap here
    * Resources shared
        6. [Messaging in Jupyter](https://jupyter-client.readthedocs.io/en/latest/messaging.html)
        7. [Jeremy Tuloup, Johan Mabille Navigating the Jupyter Landscape | JupyterCon 2023](https://www.youtube.com/watch?v=uWJ0-OPKTxI)
* Context for [Support kernel_info request on the control channel](https://github.com/jupyter/enhancement-proposals/pull/82)
    * Five channels between kernel and server: heartbeat, IO pub, stdin (kernel asks for input), shell, and control
    * When you want to execute code, you send it on the shell
    * When connecting to a kernel, you send a `kernel_info` request on the shell; but if the shell is busy, e.g. paused on a breakpoint or executing a long job, you have to wait for a long time
* Discuss [JEP for copy-to-globals support in debugger_info #93](https://github.com/jupyter/enhancement-proposals/pull/93)
* Discussed creating a kernel roadmap for JEPs voted on
    * Which version of the protocol *requires* a feature
    * Which version of the protocol is *required* by clients such as jupyter client
    * What happens in case of mis-communication of the supported protocol
    * The use of jupyter-kernel-tests as the main testing tool
* Major efforts that the SSC should take on:
    * Protocol clarification, hardening, and documentation
    * Incubation process
    * Schema publishing
